pkg/distributor/distributor_retry.go

```diff
@@ -0,0 +1,38 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package distributor
+
+import (
+	"flag"
+	"time"
+
+	"github.com/pkg/errors"
+)
+
+var (
+	errRetryBaseLessThanOneSecond    = errors.New("retry base duration should not be less than 1 second")
+	errNonPositiveMaxAllowedAttempts = errors.New("maxAllowedAttempts should be a positive value")
+)
+
+type RetryConfig struct {
+	Enabled            bool          `yaml:"enabled" category:"experimental"`
+	Base               time.Duration `yaml:"base" category:"experimental"`
+	MaxAllowedAttempts int           `yaml:"max_allowed_attempts" category:"experimental"`
+}
+
+// RegisterFlags adds the flags required to config this to the given FlagSet.
+func (cfg *RetryConfig) RegisterFlags(f *flag.FlagSet) {
+	f.BoolVar(&cfg.Enabled, "distributor.retry-after-header.enabled", false, "Enabled controls inclusion of the Retry-After header in the response: true includes it for client retry guidance, false omits it.")
+	f.DurationVar(&cfg.Base, "distributor.retry-after-header.base", 3*time.Second, "Base duration for calculating the Retry-After header in responses to 429/5xx errors.")
```

Comment:
Since `Retry-After` header only works with seconds, instead of using duration, I'd call it `base-seconds` and only allow integers. (This also saves us conversions when doing the computation).

Reply:
i can't change the variable name because of linter failed on `don't use unit-specific suffix "Seconds" (ST1011)go-staticcheck` will still update the config name

Reply:
`ST1011` warned you because you changed the name but kept the type being `time.Duration`.  If the variable is `BaseSeconds`, the type should be `int` instead.

Reply:
Now i feel terrible about not trusting ST1011, apologies